scripts/ci-build-kubernetes.sh

```diff
@@ -73,6 +73,9 @@ setup() {
     # ref: https://github.com/kubernetes/kubernetes/blob/5491484aa91fd09a01a68042e7674bc24d42687a/build/lib/release.sh#L345-L346
     export IMAGE_TAG="${KUBE_GIT_VERSION/+/_}"
     echo "using IMAGE_TAG=${IMAGE_TAG}"
+    local git=(git --work-tree "${KUBE_ROOT}")
```

Comment:
The above git pattern is copied exactly from upstream k/k, and the above is executed during every relevant prow job where we version + build k8s:

- https://github.com/kubernetes/kubernetes/blob/v1.25.4/hack/lib/version.sh#L56

So I think there's a good bet that we'll be doing this in the same operational paradigm as the exisiting tests, and this should work.

This foo did work when I tried it in k/k, but I think rather than advocate that we do it there it's probably better to do it in our own test flow:

- https://github.com/kubernetes/kubernetes/pull/114166/files#diff-8da0fd720cd36d5a6a782f2388b765e7a7028e62c9a677129929b750b3711686R63-R64

Reply:
Ok, good enough for me, thanks 👍

Reply:
It didn't work. :/

It's printing the diff from capz, not k/k. Ref:

https://prow.k8s.io/log?container=test&id=1598004180683853824&job=pull-kubernetes-e2e-capz-conformance